### PR TITLE
Add support for "SELECT ... FOR UPDATE" #224

### PR DESCRIPTION
**Pull request by jessesleeping**
Modify the logic in transaction manager layer to support "select for update". Also add a new test case.

### CODE REVIEW

File: src/concurrency/timestamp_ordering_transaction_manager.cpp [hunk 1]
```diff
@@ -357,6 +374,25 @@ bool TimestampOrderingTransactionManager::PerformRead(
...
   auto tile_group_header = tile_group->GetHeader();
 
+  // Check if it's select for update before we check the ownership and modify the
+  // last reader tid
+  if (acquire_ownership == true
+      && IsOwner(current_txn, tile_group_header, tuple_id) == false) {
+    // Acquire ownership if we haven't
+    if (IsOwnable(current_txn, tile_group_header, tuple_id) == false) {
+      // Can not own
+      printf("[JX] Read Own Txn IsOwnerable failed\n");
```
Comment (jessesleeping) [inline src/concurrency/timestamp_ordering_transaction_manager.cpp]: Sorry I forgot to remove it .....

**Comment by malin1993ml** [comment]
There're two PRs that merged into master today. Can you rebase your change on top of the current master?

Also, can you run TPC-C through OLTPBench to make sure this doesn't break TPC-C before we merge it?

**Comment by jessesleeping** [comment]
Sure, I can do the rebase.

For the TPC-C, currently I can not run it on our master branch. It seems like there is a SQL parse error when creating the table. Also, I think it would be better if @ranxian or @yingjunwu can review my code before we merge it...

**Comment by malin1993ml** [comment]
Okay. Sounds reasonable.

**Pull request merged**